src/Cooked/Pretty/Cooked.hs

```diff
@@ -214,19 +219,28 @@ prettySigners _ _ [] = []
 -- Examples without and with redeemer
 -- > #abcdef "Foo" -> 500
 -- > #123456 "Bar" | Redeemer -> 1000
-prettyMints :: PrettyCookedOpts -> (Script.Versioned Script.MintingPolicy, MintsRedeemer, Api.TokenName, Integer) -> DocCooked
-prettyMints opts (policy, NoMintsRedeemer, tokenName, amount) =
+prettyMints :: PrettyCookedOpts -> (Script.Versioned Script.MintingPolicy, TxSkelRedeemer, Api.TokenName, Integer) -> DocCooked
+prettyMints opts (policy, TxSkelNoRedeemer, tokenName, amount) =
   prettyCookedOpt opts policy
     <+> PP.viaShow tokenName
     <+> "->"
     <+> PP.viaShow amount
-prettyMints opts (policy, SomeMintsRedeemer redeemer, tokenName, amount) =
+prettyMints opts (policy, TxSkelRedeemerForScript redeemer, tokenName, amount) =
   prettyCookedOpt opts policy
     <+> PP.viaShow tokenName
     <+> "|"
     <+> prettyCookedOpt opts redeemer
     <+> "->"
     <+> PP.viaShow amount
+prettyMints opts (policy, TxSkelRedeemerForReferencedScript oref redeemer, tokenName, amount) =
+  prettyCookedOpt opts policy
+    <+> PP.viaShow tokenName
+    <+> "|"
+    <+> prettyCookedOpt opts redeemer
+    <+> " (with reference script at "
+    <+> prettyCookedOpt opts oref
+    <+> ") ->"
+    <+> PP.viaShow amount
```

Comment:
Might become a bit long and unreadable when we have long token names.
Just making a note of it to remember, this is good in this PR.